src/Api/Response/ResponseParser.php

```diff
@@ -51,171 +53,121 @@ public function parse(HttpRequest $request, $responseData): Response
     {
-        if (!isset($data->landingPage)
-            || empty($data->landingPage->attributes())
-            || !isset($data->landingPage->attributes()->link)
-        ) {
-            return null;
-        }
-
-        return $data->landingPage->attributes()->link->__toString();
+        return $data['metadata']['landingpage'] ?: null;
     }
 
     /**
-     * @param SimpleXMLElement $data
+     * @param array $data
      * @return array
      */
-    protected function parsePromotion(SimpleXMLElement $data)
+    protected function parsePromotion(array $data): array
     {
         $promotion = [];
 
-        if (isset($data->promotion) && !empty($data->promotion->attributes())) {
-            $promotion['image'] = $data->promotion->attributes()->image->__toString();
-            $promotion['link'] = $data->promotion->attributes()->link->__toString();
+        if (isset($data['metadata']['promotion'])) {
+            $promotion['image'] = $data['metadata']['promotion']['imageUrl'];
+            $promotion['link'] = $data['metadata']['promotion']['url'];
         }
 
         return $promotion;
     }
 
     /**
-     * @param SimpleXMLElement $data
+     * @param array $metadata
      * @return array
      */
-    protected function parseResults(SimpleXMLElement $data)
+    protected function parseResults(array $metadata): array
     {
         $results = [];
 
-        if (!empty($data->results)) {
-            $results['count'] = $data->results->count->__toString();
+        if (isset($metadata['totalResults'])) {
+            $results['count'] = $metadata['totalResults'];
         }
 
         return $results;
     }
 
     /**
-     * @param SimpleXMLElement $data
+     * @param array $data
      * @return array
      */
-    protected function parseProducts(SimpleXMLElement $data)
+    protected function parseProducts(array $data): array
     {
-        $products = [];
+        return $data['items'] ?: [];
+    }
 
-        if (!empty($data->products)) {
-            foreach ($data->products->product as $product) {
-                $productData = [
-                    'id' => $product['id']->__toString(),
-                    'relevance' => $product['relevance']->__toString(),
-                ];
+    protected function parseQueryInfoMessage(HttpRequest $request, array $data): array
+    {
+        @list('didYouMeanQuery' => $didYouMeanQuery, 'improvedQuery' => $improvedQuery, 'correctedQuery' => $correctedQuery) = $data['result']['variant'];
+        
+        $currentQuery = $data['result']['metadata']['effectiveQuery'] ?: null;
 
-                foreach ($product->properties->property as $property) {
-                    $productData['properties'][$property['name']->__toString()] = $property->__toString();
-                }
+        $queryStringType = null;
 
-                $products[] = $productData;
-            }
+        if($improvedQuery){
+            $queryStringType = 'improved';
```

Comment:
@rimvydas-hashtages Only one more thing: Please let me know why we never need to set the `didYouMean` type.

Reply:
Because old code always overrides the type to 'did-you-mean' whenever the 'didYouMeanQuery' is set:
![image](https://github.com/findologic/plugin-plentymarkets-ceres-api/assets/83945230/8f2d5753-9e24-4b7e-aeb2-8f809fa62c9c)
![image](https://github.com/findologic/plugin-plentymarkets-ceres-api/assets/83945230/27c3a9b1-9f49-4f75-9d94-9af4c88d5c06)